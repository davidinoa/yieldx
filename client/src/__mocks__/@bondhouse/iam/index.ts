import {
  CreateInvestorMappingRequest,
  UserType,
  UserStatus
} from "@bondhouse/iam";

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function UsersApi() {}

UsersApi.prototype.getCurrentUser = jest.fn(() =>
  Promise.resolve({
    data: {
      user: {
        id: "auth0|5ec6119474dfa00cbf1d53cb",
        first: "Institution",
        last: "New User",
        userType: "ADMIN",
        email: "test4@bondhouse.com",
        terms: true,
        institutionId: "bf8a5597-9569-4510-babd-a6d9655e9f0d",
        createdAt: "2020-05-21T05:28:52.982380Z",
        updatedAt: "2020-05-21T05:28:52.982380Z",
        status: "ACTIVE",
        policies: []
      }
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function PasswordResetTicketsApi() {
  return {};
}

PasswordResetTicketsApi.prototype.createPasswordResetTicket = jest.fn(() =>
  Promise.resolve({
    data: {
      success: true
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInventoryMappingApi() {}

InstitutionInventoryMappingApi.prototype.getInstitutionInventoryMapping = jest.fn(
  () =>
    Promise.resolve({
      data: {
        inventoryMapping: {
          institutionId: "b17c644a-30e8-47fb-a26b-f6743786c3ba",
          inpaas: "inpaas",
          bestFit: "bestfit",
          assetExplorer: "assetexplorer"
        }
      }
    })
);
InstitutionInventoryMappingApi.prototype.createOrUpdateInstitutionInventoryMapping = jest.fn();

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInvestorMappingApi() {}

InstitutionInvestorMappingApi.prototype.createInstitutionInvestorMapping = jest.fn(
  (
    institutionId: string,
    createInvestorMappingRequest?: CreateInvestorMappingRequest
  ) =>
    Promise.resolve({
      data: {
        investorMapping: {
          institutionId,
          investorId: createInvestorMappingRequest?.investorId || ""
        }
      }
    })
);

export { UserType, UserStatus };
