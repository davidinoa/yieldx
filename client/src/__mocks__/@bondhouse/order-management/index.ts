import {
  OrderStatus,
  CreateOrderRequestDirectionEnum
} from "@bondhouse/order-management";
import { openOrder } from "containers/Portfolios/PortfolioDisplay/OrdersAndTrades/OrdersAndTrades.test";

// eslint-disable-next-line @typescript-eslint/no-empty-function
function OrdersApi() {}

OrdersApi.prototype.getOrders = jest.fn(() => {
  return Promise.resolve({
    data: {
      orders: []
    }
  });
});
OrdersApi.prototype.createOrder = jest.fn(() => {
  return Promise.resolve({
    data: {
      orders: [openOrder]
    }
  });
});
export { OrdersApi, OrderStatus, CreateOrderRequestDirectionEnum };
