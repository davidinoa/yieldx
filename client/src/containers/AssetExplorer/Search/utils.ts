interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
