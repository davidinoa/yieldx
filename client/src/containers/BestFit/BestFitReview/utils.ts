import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
