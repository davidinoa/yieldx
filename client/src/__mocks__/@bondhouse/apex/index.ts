import { AccountStatus } from "@bondhouse/apex";
import { MOCK_APEX_ACCOUNT } from "./__fixtures__";
// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function ApexAccountsApi() {}

ApexAccountsApi.prototype.getApexAccountByInvestorId = jest.fn(() =>
  Promise.resolve({
    data: {
      apexAccount: MOCK_APEX_ACCOUNT
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function StatementsApi() {}

StatementsApi.prototype.getAccountStatements = jest.fn(() =>
  Promise.resolve([
    {
      apexAccountId: "28b5b427-053b-4ad8-9acb-05a5e7bbe614",
      statmentDate: "12/31/2014",
      documentType: "Account Statement",
      statementURL:
        "https://UATEDocuments.apexclearing.com/RedirectConfirmURL.aspx?ds=1DC69CF82426F431D4535F6549270DAC58EEB05D61A8209E49CAAE3BC76EE8B866CEDC5A04A68284717045A7352477CC49D5E186BA7CA707C0EF8A692C654BED5732532662D182181AAF4E3C7F5B219956BF0F1B6F57C588C0465DF32D3A5EB58EFB4CFF188B33275E8BFB631B97C80DFEB0838AECFC00DF537E96EB28868EFF10660E83DD1A6C5808E6DED036C15E25336F448E9F69DF30118E6960DD1F98D4B8E4714BCAE3D1435A79505BC3FB730B09FEA86B8E3DEB1F8DDAF65E746FBE833308839C1FA0381057F4E71EFC7E14AF5DBB327B2CA848EC4CC2D036AF66C648E78AD70688C550DEB37F942CC82AEEE4F6E1BA95C082DB8D326BC86D5B5DC113060E7449A97FD2F019A66050CA0AF83B0FA50FFCFE9F5E1664C92DA902988BDF&pid=TkVXU1RNVHx2NzEyNi01Mj9yLTUyOTMtNzc1xS1PREEyLTI4MzBGQUFZLTM5NzAxLTU0MzIxNC00MjE4NzItMjE5MjAyLTg1LTc5LTI2NDMtODktMC1eATAxNAE1UEkBMzIyNzIBTUFSSyBTTEFWT05JQQExN1EwNgFXRUEBAgECAQIBAgExNDQBMTcxMDgBMTMxODQBAgECAQIBA0101816"
    }
  ])
);

export { ApexAccountsApi, AccountStatus, StatementsApi };
