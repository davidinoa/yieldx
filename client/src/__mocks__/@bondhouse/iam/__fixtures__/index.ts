import { User } from "@bondhouse/iam";
import { UserType } from "..";

export const mockAdminUser: User = {
  id: "id-admin",
  email: "mock.admin@bondhouse.com",
  first: "Mock",
  last: "Admin",
  userType: UserType.ADMIN,
  terms: true
};
