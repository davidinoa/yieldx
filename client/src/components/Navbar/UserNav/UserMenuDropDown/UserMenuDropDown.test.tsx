import React from "react";
import { User, UserType } from "@bondhouse/iam";
import dayjs from "dayjs";
import UserDropDown from "components/Navbar/UserNav/UserMenuDropDown/UserMenuDropDown";
import { render, screen } from "test/utils";

dayjs.tz.setDefault("America/New_York");

describe("UserMenuDropDown", () => {
  test("Should render", () => {
    const mockUser: User = {
      id: "",
      email: "",
      first: "Mock",
      last: "User",
      userType: UserType.ADMIN,
      terms: true
    };
    const { container } = render(
      <UserDropDown
        user={mockUser}
        toggleDropDown={jest.fn}
        lastLogin={new Date()}
      />
    );
    expect(container).toBeDefined();
  });

  test("Admin user type should see manage users and investor accounts links", () => {
    const mockUser: User = {
      id: "",
      email: "",
      first: "Mock",
      last: "User",
      userType: UserType.ADMIN,
      terms: true,
      institutionId: "mock-institution"
    };
    const { asFragment } = render(
      <UserDropDown
        user={mockUser}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Investor accounts/i)).toBeInTheDocument();
    expect(screen.queryByText(/Manage users/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Default user type should not see manage users link", () => {
    const mockUser: User = {
      id: "",
      email: "",
      first: "Mock",
      last: "User",
      userType: UserType.DEFAULT,
      terms: true,
      institutionId: "mock-institution"
    };
    const { asFragment } = render(
      <UserDropDown
        user={mockUser}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Investor accounts/i)).toBeInTheDocument();
    expect(screen.queryByText(/Manage users/i)).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Trial user type should not see manage users link or investors link", () => {
    const mockUser: User = {
      id: "",
      email: "",
      first: "Mock",
      last: "User",
      userType: UserType.TRIAL,
      terms: true,
      institutionId: "mock-institution"
    };
    const { asFragment } = render(
      <UserDropDown
        user={mockUser}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Investor accounts/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Manage users/i)).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Manage users should only appear if user is ADMIN and has non default institution", () => {
    const mockUser: User = {
      id: "",
      email: "",
      first: "Mock",
      last: "User",
      userType: UserType.ADMIN,
      terms: true
    };
    const { rerender } = render(
      <UserDropDown
        user={mockUser}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Manage users/i)).not.toBeInTheDocument();

    rerender(
      <UserDropDown
        user={{ ...mockUser, institutionId: "bondhouse" }}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Manage users/i)).not.toBeInTheDocument();
    rerender(
      <UserDropDown
        user={{ ...mockUser, institutionId: "mock-id" }}
        toggleDropDown={jest.fn}
        lastLogin="2018-04-04T16:00:00.000Z"
      />
    );
    expect(screen.queryByText(/Manage users/i)).toBeInTheDocument();
  });
});
