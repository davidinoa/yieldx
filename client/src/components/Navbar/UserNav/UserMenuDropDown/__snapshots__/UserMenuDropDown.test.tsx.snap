// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenuDropDown Admin user type should see manage users and investor accounts links 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  margin: 1rem 0 0;
  min-width: 256px;
  height: auto;
  border: 1px solid var(--border-new);
  border-radius: 5px;
  z-index: 10000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.live-header-options {
  top: 50%;
  right: 0;
  -webkit-transform: translate(1%,2%);
  -ms-transform: translate(1%,2%);
  transform: translate(1%,2%);
}

.c0.user-options {
  top: 54%;
  left: 7%;
  -webkit-transform: translate(-83%,3%);
  -ms-transform: translate(-83%,3%);
  transform: translate(-83%,3%);
}

.c0 > div.arrow {
  width: 60px;
  height: 16px;
  overflow: hidden;
  position: absolute;
  right: 0;
  margin-right: 0;
  top: -15px;
}

.c0 > div.arrow:after {
  content: "";
  position: absolute;
  left: 20px;
  top: 5px;
  width: 20px;
  height: 20px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--dark);
  border: 1px solid var(--border-new);
  border-radius: 2px;
}

.c1 {
  width: 100%;
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 5px 5px 0px 0px;
  background: var(--dark);
  border-bottom: 1px solid var(--border-new);
  box-sizing: border-box;
  padding: 1.5rem 20px;
}

.c1 .user-name {
  display: block;
  width: 100%;
  font-weight: bold;
  font-size: 15px;
  line-height: 18px;
  text-align: left;
}

.c1 div.user-last-login * {
  color: var(--subtext-3);
}

.c1 div .last-login-label {
  font-weight: bold;
  font-size: 10px;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  text-transform: uppercase;
  margin-right: 0.25rem;
  display: inline-block;
  padding-top: 0.185rem;
}

.c1 div .last-login-value {
  font-family: Roboto Mono;
  font-size: 11px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: var(--dark);
  border-radius: 0 0 5px 5px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 .sign-out-container {
  color: var(--primary);
  height: 2rem;
  font-weight: bold;
  font-size: 14px;
  padding: 1.5rem 20px;
}

.c3 {
  width: 100%;
  height: 45px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  opacity: initial;
  padding-left: 20px;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 14px;
  line-height: 17px;
  color: white;
}

.c3:hover {
  background: var(--medium);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c3.manage-users {
  border-top: 1px solid var(--border-new);
  border-bottom: 1px solid var(--border-new);
}

@media (min-width:1600px) {
  .c0.live-header-options {
    -webkit-transform: translate(18%,1%);
    -ms-transform: translate(18%,1%);
    transform: translate(18%,1%);
  }
}

@media (min-width:1600px) {
  .c0.user-options {
    -webkit-transform: translate(-65%,4%);
    -ms-transform: translate(-65%,4%);
    transform: translate(-65%,4%);
  }
}

@media (min-width:1600px) {
  .c0 > div.arrow {
    margin-right: 35px;
    width: 70px;
  }
}

<div
    class="c0 user-options"
  >
    <div
      class="arrow"
    />
    <div
      class="c1"
    >
      <span
        class="user-name"
      >
        Mock User
      </span>
      <div
        class="user-last-login d-flex justify-content-center align-items-center"
      >
        <span
          class="last-login-label"
        >
          Last login: 
        </span>
        <span
          class="last-login-value"
        >
          Wed Apr 04 @4:00pm
        </span>
      </div>
    </div>
    <div
      class="c2"
    >
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Login-&-security"
        href="/login-security"
      >
        Login & security
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Investor-accounts"
        href="/investors"
      >
        Investor accounts
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Portfolios"
        href="/portfolios"
      >
        Portfolios
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Favorites"
        href="/favorites"
      >
        Favorites
      </a>
      <a
        class="c3 nav-item manage-users"
        data-testid="Nav-Item-Manage-users"
        href="/users"
      >
        Manage users
      </a>
      <div
        class="sign-out-container d-flex w-100 align-items-center justify-content-center"
      >
        <a
          data-testid="logout-link"
          href="#"
        >
          Log Out
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserMenuDropDown Default user type should not see manage users link 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  margin: 1rem 0 0;
  min-width: 256px;
  height: auto;
  border: 1px solid var(--border-new);
  border-radius: 5px;
  z-index: 10000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.live-header-options {
  top: 50%;
  right: 0;
  -webkit-transform: translate(1%,2%);
  -ms-transform: translate(1%,2%);
  transform: translate(1%,2%);
}

.c0.user-options {
  top: 54%;
  left: 7%;
  -webkit-transform: translate(-83%,3%);
  -ms-transform: translate(-83%,3%);
  transform: translate(-83%,3%);
}

.c0 > div.arrow {
  width: 60px;
  height: 16px;
  overflow: hidden;
  position: absolute;
  right: 0;
  margin-right: 0;
  top: -15px;
}

.c0 > div.arrow:after {
  content: "";
  position: absolute;
  left: 20px;
  top: 5px;
  width: 20px;
  height: 20px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--dark);
  border: 1px solid var(--border-new);
  border-radius: 2px;
}

.c1 {
  width: 100%;
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 5px 5px 0px 0px;
  background: var(--dark);
  border-bottom: 1px solid var(--border-new);
  box-sizing: border-box;
  padding: 1.5rem 20px;
}

.c1 .user-name {
  display: block;
  width: 100%;
  font-weight: bold;
  font-size: 15px;
  line-height: 18px;
  text-align: left;
}

.c1 div.user-last-login * {
  color: var(--subtext-3);
}

.c1 div .last-login-label {
  font-weight: bold;
  font-size: 10px;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  text-transform: uppercase;
  margin-right: 0.25rem;
  display: inline-block;
  padding-top: 0.185rem;
}

.c1 div .last-login-value {
  font-family: Roboto Mono;
  font-size: 11px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: var(--dark);
  border-radius: 0 0 5px 5px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 .sign-out-container {
  color: var(--primary);
  height: 2rem;
  font-weight: bold;
  font-size: 14px;
  padding: 1.5rem 20px;
}

.c3 {
  width: 100%;
  height: 45px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  opacity: initial;
  padding-left: 20px;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 14px;
  line-height: 17px;
  color: white;
}

.c3:hover {
  background: var(--medium);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c3.manage-users {
  border-top: 1px solid var(--border-new);
  border-bottom: 1px solid var(--border-new);
}

@media (min-width:1600px) {
  .c0.live-header-options {
    -webkit-transform: translate(18%,1%);
    -ms-transform: translate(18%,1%);
    transform: translate(18%,1%);
  }
}

@media (min-width:1600px) {
  .c0.user-options {
    -webkit-transform: translate(-65%,4%);
    -ms-transform: translate(-65%,4%);
    transform: translate(-65%,4%);
  }
}

@media (min-width:1600px) {
  .c0 > div.arrow {
    margin-right: 35px;
    width: 70px;
  }
}

<div
    class="c0 user-options"
  >
    <div
      class="arrow"
    />
    <div
      class="c1"
    >
      <span
        class="user-name"
      >
        Mock User
      </span>
      <div
        class="user-last-login d-flex justify-content-center align-items-center"
      >
        <span
          class="last-login-label"
        >
          Last login: 
        </span>
        <span
          class="last-login-value"
        >
          Wed Apr 04 @4:00pm
        </span>
      </div>
    </div>
    <div
      class="c2"
    >
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Login-&-security"
        href="/login-security"
      >
        Login & security
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Investor-accounts"
        href="/investors"
      >
        Investor accounts
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Portfolios"
        href="/portfolios"
      >
        Portfolios
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Favorites"
        href="/favorites"
      >
        Favorites
      </a>
      <div
        class="sign-out-container d-flex w-100 align-items-center justify-content-center"
      >
        <a
          data-testid="logout-link"
          href="#"
        >
          Log Out
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserMenuDropDown Trial user type should not see manage users link or investors link 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  margin: 1rem 0 0;
  min-width: 256px;
  height: auto;
  border: 1px solid var(--border-new);
  border-radius: 5px;
  z-index: 10000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.live-header-options {
  top: 50%;
  right: 0;
  -webkit-transform: translate(1%,2%);
  -ms-transform: translate(1%,2%);
  transform: translate(1%,2%);
}

.c0.user-options {
  top: 54%;
  left: 7%;
  -webkit-transform: translate(-83%,3%);
  -ms-transform: translate(-83%,3%);
  transform: translate(-83%,3%);
}

.c0 > div.arrow {
  width: 60px;
  height: 16px;
  overflow: hidden;
  position: absolute;
  right: 0;
  margin-right: 0;
  top: -15px;
}

.c0 > div.arrow:after {
  content: "";
  position: absolute;
  left: 20px;
  top: 5px;
  width: 20px;
  height: 20px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--dark);
  border: 1px solid var(--border-new);
  border-radius: 2px;
}

.c1 {
  width: 100%;
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 5px 5px 0px 0px;
  background: var(--dark);
  border-bottom: 1px solid var(--border-new);
  box-sizing: border-box;
  padding: 1.5rem 20px;
}

.c1 .user-name {
  display: block;
  width: 100%;
  font-weight: bold;
  font-size: 15px;
  line-height: 18px;
  text-align: left;
}

.c1 div.user-last-login * {
  color: var(--subtext-3);
}

.c1 div .last-login-label {
  font-weight: bold;
  font-size: 10px;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  text-transform: uppercase;
  margin-right: 0.25rem;
  display: inline-block;
  padding-top: 0.185rem;
}

.c1 div .last-login-value {
  font-family: Roboto Mono;
  font-size: 11px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background-color: var(--dark);
  border-radius: 0 0 5px 5px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 .sign-out-container {
  color: var(--primary);
  height: 2rem;
  font-weight: bold;
  font-size: 14px;
  padding: 1.5rem 20px;
}

.c3 {
  width: 100%;
  height: 45px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  opacity: initial;
  padding-left: 20px;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 14px;
  line-height: 17px;
  color: white;
}

.c3:hover {
  background: var(--medium);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c3.manage-users {
  border-top: 1px solid var(--border-new);
  border-bottom: 1px solid var(--border-new);
}

@media (min-width:1600px) {
  .c0.live-header-options {
    -webkit-transform: translate(18%,1%);
    -ms-transform: translate(18%,1%);
    transform: translate(18%,1%);
  }
}

@media (min-width:1600px) {
  .c0.user-options {
    -webkit-transform: translate(-65%,4%);
    -ms-transform: translate(-65%,4%);
    transform: translate(-65%,4%);
  }
}

@media (min-width:1600px) {
  .c0 > div.arrow {
    margin-right: 35px;
    width: 70px;
  }
}

<div
    class="c0 user-options"
  >
    <div
      class="arrow"
    />
    <div
      class="c1"
    >
      <span
        class="user-name"
      >
        Mock User
      </span>
      <div
        class="user-last-login d-flex justify-content-center align-items-center"
      >
        <span
          class="last-login-label"
        >
          Last login: 
        </span>
        <span
          class="last-login-value"
        >
          Wed Apr 04 @4:00pm
        </span>
      </div>
    </div>
    <div
      class="c2"
    >
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Login-&-security"
        href="/login-security"
      >
        Login & security
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Portfolios"
        href="/portfolios"
      >
        Portfolios
      </a>
      <a
        class="c3 nav-item"
        data-testid="Nav-Item-Favorites"
        href="/favorites"
      >
        Favorites
      </a>
      <div
        class="sign-out-container d-flex w-100 align-items-center justify-content-center"
      >
        <a
          data-testid="logout-link"
          href="#"
        >
          Log Out
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
